cmd/kms-rest/startcmd/start.go

```diff
@@ -37,20 +40,61 @@ import (
 )
 
 const (
+	commonEnvVarUsageText = "Alternatively, this can be set with the following environment variable: "
+
 	hostURLFlagName      = "host-url"
 	hostURLFlagShorthand = "u"
 	hostURLFlagUsage     = "URL to run the kms-rest instance on. Format: HostName:Port."
 	hostURLEnvKey        = "KMS_REST_HOST_URL"
 
+	databaseTypeFlagName      = "database-type"
+	databaseTypeEnvKey        = "DATABASE_TYPE"
+	databaseTypeFlagShorthand = "t"
+	databaseTypeFlagUsage     = "The type of database to use for storing metadata about keystores and" +
+		"associated keys. Supported options: mem, couchdb. " + commonEnvVarUsageText + databaseTypeEnvKey
+
+	databaseURLFlagName      = "database-url"
```

Comment:
What's the non-secrets database going to be used for?

Reply:
It's used for storing metadata related to the keystore (like owner) and list of associated keys. In future it might store some additional data like zcap and related authorization capabilities.

Reply:
Does it need to be a separate store?

Reply:
I believe that yes. We use them for a different purpose. The one for KMS is "blindly" passed to the Aries KMS Creator to use for storing key primitives. I don't want to mess with it. Another one is the application-specific that serves keystore needs. You can use completely different backends for them, not least for security considerations.